spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/wavefront/WavefrontMetricsExportAutoConfiguration.java

```diff
@@ -49,29 +45,17 @@
  */
 @AutoConfiguration(
 		before = { CompositeMeterRegistryAutoConfiguration.class, SimpleMetricsExportAutoConfiguration.class },
-		after = MetricsAutoConfiguration.class)
-@ConditionalOnBean(Clock.class)
-@ConditionalOnClass({ WavefrontMeterRegistry.class, WavefrontSender.class })
+		after = { MetricsAutoConfiguration.class, WavefrontAutoConfiguration.class })
+@ConditionalOnBean({ Clock.class, WavefrontSender.class })
+@ConditionalOnClass(WavefrontMeterRegistry.class)
```

Comment:
I wonder if this will break if `WavefrontMeterRegistry` is on the classpath but `WavefrontSender` is not. The previous conditions seemed to be protecting against that.

Reply:
Doesn't `@ConditionalOnBean` include `@ConditionalOnClass`? This configuration only gets activated when there's a `WavefrontSender`, which means it must be on the classpath. Or is there some case I'm not seeing?

Reply:
It does, but in a somewhat complex manner. The two conditions can also be processed in different phases, with `@ConditionalOnClass` going first. What you have now may well work, but I'd find it easier to understand what's going on if the class condition for `WavefrontSender` was still in place.